src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -38,37 +39,58 @@ const propTypes = {
         /** The authToken for the current session */
         email: PropTypes.string,
     }),
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
 };
 
 const defaultProps = {
     route: {
         params: {},
     },
     session: {},
+    betas: null,
 };
+
 class LogInWithShortLivedTokenPage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {run: false};
+    }
+
     componentDidMount() {
         const accountID = parseInt(lodashGet(this.props.route.params, 'accountID', ''), 10);
         const email = lodashGet(this.props.route.params, 'email', '');
         const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
         const encryptedAuthToken = lodashGet(this.props.route.params, 'encryptedAuthToken', '');
 
-        // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
-        const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
-
-        // If the user is revisiting the component authenticated or they were already logged into the right account, we simply redirect them to the exitTo
+        // If the user is revisiting the component authenticated or they were already logged into
+        // the right account, we need to ensure they have the betas loaded, then in componentDidUpdate we will
         if (this.props.session.authToken && email === this.props.session.email) {
-            // In order to navigate to a modal, we first have to dismiss the current modal. But there is no current
-            // modal you say? I know, it confuses me too. Without dismissing the current modal, if the user cancels out
-            // of the workspace modal, then they will be routed back to
-            // /transition/<accountID>/<email>/<authToken>/workspace/<policyID>/card and we don't want that. We want them to go back to `/`
-            // and by calling dismissModal(), the /transition/... route is removed from history so the user will get taken to `/`
-            // if they cancel out of the new workspace modal.
-            Navigation.dismissModal();
-            Navigation.navigate(exitTo);
+            User.getBetas();
```

Comment:
So this is what triggers the beta update. But this is an async function which will complete AFTER state has been set to `{run: true}`

Reply:
Maybe I misunderstand, but I thought one of the original bugs was that betas hadn't yet been retrieved?!

Reply:
ok, this is exactly what tripped me and confused me when writing this. 
So, there's really 2 flows here:
- You are not logged in
- You are already logged in

If you are already logged in, all we need to do is to ensure we call `User.getBetas` so onyx re-renders this and `componentDidUpdate` can safely navigate to the page.

If you are not logged in, you can't even load the betas yet (since that's an authenticated API call), so instead we `signinWithShortLivedToken`, which after having signed in, will load the betas and then same as when you are logged in onyx re-renders this and `componentDidUpdate` can safely navigate to the page.

Reply:
Yep thanks. I also missed that we return after calling `User.getBetas` and thought that we would set state to hasRan before the betas API call completes